portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,25 +14,107 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Enumeration;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-//    Arraylist<String> comments = new Arraylist<String>();
-//     comments.add("I am a pretty girl");
-//     comments.add("I am a cute girl");
-//     comments.add("I am a phenomenal girl");
-
-  String comment = "Hello Chidera!";
-
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println(comment);
-  }
-}
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+        Query query = new Query("CommentData").addSort("timestamp", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query); 
+      
+        String lang = request.getParameter("lang");
+
+      
+        List<CommentData> commentlog = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String userName = (String) entity.getProperty("userName");
+            String userComment = (String) entity.getProperty("userComment");
+            long timestamp = (long) entity.getProperty("timestamp");
+
+            String Comment = translateComment(userComment, lang);
+
+
+            CommentData task = new CommentData(id, timestamp, userName, Comment);
+
+            commentlog.add(task);
+        }
+
+
+        // Convert the comments to JSON
+        String json = convertToJsonUsingGson(commentlog);
+
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+  
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String userComment = request.getParameter("userComment");
+        String userName = request.getParameter("userName");
+        long timestamp = System.currentTimeMillis();
+
+        Entity commentEntity = new Entity("CommentData");
+        commentEntity.setProperty("userName", userName);
+        commentEntity.setProperty("userComment", userComment);
+        commentEntity.setProperty("timestamp", timestamp);
+        
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
+
+        // Redirect back to the HTML page.
+        response.sendRedirect("/index.html");
+    }
+    
+    private String convertToJsonUsingGson(List<CommentData> commentlog) {
+        Gson gson = new Gson();
+        String json = gson.toJson(commentlog);
+        return json;
```

Comment:
You can probably combine this with the line above
`return gson.toJson(commentLog)`